From b110595e780e2f0c659021ca77b20693e250d91c Mon Sep 17 00:00:00 2001
From: "brian.zhu" <brian.zhu@amlogic.com>
Date: Mon, 18 Aug 2014 16:53:43 +0800
Subject: [PATCH 4727/5965] PD#96047 fix verifier for camera video

1.re-order canvas index.
2.seperate the ion device memory from ppmgr memory

Change-Id: If2d786e4b2ffebd0003f398038648106e8e002ba
---
 arch/arm/boot/dts/amlogic/meson8b_m102_1G.dtd |  5 ++---
 include/linux/amlogic/amports/canvas.h        | 12 ++++++------
 2 files changed, 8 insertions(+), 9 deletions(-)

diff --git a/arch/arm/boot/dts/amlogic/meson8b_m102_1G.dtd b/arch/arm/boot/dts/amlogic/meson8b_m102_1G.dtd
index fccc06023d99..0507889be86b 100755
--- a/arch/arm/boot/dts/amlogic/meson8b_m102_1G.dtd
+++ b/arch/arm/boot/dts/amlogic/meson8b_m102_1G.dtd
@@ -281,9 +281,8 @@ void root_func(){
         compatible = "amlogic,ion_dev";
         dev_name = "ion_dev";
         status = "ok";
-        share-memory-name = "ppmgr0";
-        share-memory-offset = <0>;
-        share-memory-size = <0x01000000>;//ion_dev use same buffer with ppmgr0
+	reserve-memory = <0x01000000>;
+	reserve-iomap = "true";
     };
 /// ***************************************************************************************
 ///	-	DISP&MM-Vout
diff --git a/include/linux/amlogic/amports/canvas.h b/include/linux/amlogic/amports/canvas.h
index ba2a7a7c41f1..dceb6a7d9414 100755
--- a/include/linux/amlogic/amports/canvas.h
+++ b/include/linux/amlogic/amports/canvas.h
@@ -57,7 +57,7 @@ typedef struct {
 #define OSD2_CANVAS_INDEX 0x43
 #define OSD3_CANVAS_INDEX 0x41
 #define OSD4_CANVAS_INDEX 0x42
-#define ALLOC_CANVAS_INDEX  0x46
+#define ALLOC_CANVAS_INDEX  0x44
 
 #define FREESCALE_CANVAS_INDEX 0x50   //for osd&video scale use
 #define MAX_FREESCALE_CANVAS_INDEX 0x55
@@ -115,11 +115,11 @@ typedef struct {
 #endif
 
 #ifdef CONFIG_AM_MEMPROTECT
-#define DISPLAY_CANVAS_YDUP_INDEX 0x50
-#define DISPLAY_CANVAS_UDUP_INDEX 0x51
+#define DISPLAY_CANVAS_YDUP_INDEX 0x6e
+#define DISPLAY_CANVAS_UDUP_INDEX 0x6f
 #endif
 
-#define AMVIDEOCAP_CANVAS_INDEX 0x50
+#define AMVIDEOCAP_CANVAS_INDEX 0x6e
 
 #define MIPI_CANVAS_INDEX 0x70
 #define MIPI_CANVAS_MAX_INDEX 0x7f
@@ -128,8 +128,8 @@ typedef struct {
 #define VDIN_CANVAS_INDEX              0x26
 #define VDIN_CANVAS_MAX_INDEX          0x3B
 
-#define CAMERA_USER_CANVAS_INDEX             0x98
-#define CAMERA_USER_CANVAS_MAX_INDEX     0xaf
+#define CAMERA_USER_CANVAS_INDEX             0x4e
+#define CAMERA_USER_CANVAS_MAX_INDEX     0x5f
 
 #if MESON_CPU_TYPE >= MESON_CPU_TYPE_MESON8
 #define AMLVIDEO2_RES_CANVAS 0xD8
-- 
2.19.0

